blog/top-5-cybersecurity-incidents-in-cicd/index.md

```diff
@@ -0,0 +1,118 @@
+
+**Associated OWASP CI/CD Risks:**  
+- [CICD-SEC-7: Insecure System Configuration](https://owasp.org/www-project-top-10-ci-cd-security-risks/CICD-SEC-07-Insecure-System-Configuration)  
+- [CICD-SEC-9: Improper Artifact Integrity Validation](https://owasp.org/www-project-top-10-ci-cd-security-risks/CICD-SEC-09-Improper-Artifact-Integrity-Validation)
+
+**References:**  
+- [PHP Git server server compromised, move to GitHub, and delayed updates (Php Watch)](https://php.watch/news/2021/03/git-php-net-hack)  
+
+---
+
+### **5. Dependency Confusion Attacks (2021)**
+
+**How the attack happened:**  
+Security researcher Alex Birsan demonstrated a new attack method known as "dependency confusion." He discovered that many companies, including Microsoft, Apple, and dozens of others, were vulnerable due to how they managed dependencies in their CI/CD pipelines. Birsan found that some companies’ internal packages, typically stored in private repositories, were not protected against name squatting. He uploaded malicious packages with the same names as these internal dependencies to public repositories (like npm, PyPI, and RubyGems). Due to misconfigured package managers or default configurations that prioritize public over private repositories, the CI/CD systems unknowingly downloaded and integrated the malicious public packages instead of the legitimate internal ones.
+
+**Issues in the software supply chain:**  
+The attack exploited a lack of proper validation and prioritization of package sources within dependency management processes. It highlighted the risks of dependency chain abuse and the ungoverned use of third-party repositories. The incident underscored the critical need for strict controls over external packages, better package management configurations, and proper validation to prevent unauthorized package downloads.
+
+**Impact:**  
+Many affected companies had to urgently review and overhaul their dependency management strategies to prevent such vulnerabilities. This incident significantly raised awareness of the importance of verifying and securing all components within the software supply chain. It prompted widespread adoption of new best practices, such as using scoped package names, configuring package managers to prefer internal sources, and implementing tighter access controls on internal repositories.
+
+**Associated OWASP CI/CD Risks:**  
+- [CICD-SEC-3: Dependency Chain Abuse](https://owasp.org/www-project-top-10-ci-cd-security-risks/CICD-SEC-03-Dependency-Chain-Abuse)  
+- [CICD-SEC-8: Ungoverned Usage of 3rd Party Services](https://owasp.org/www-project-top-10-ci-cd-security-risks/CICD-SEC-08-Ungoverned-Usage-of-3rd-Party-Services)
+
+**References:**  
+- [Dependency Confusion: How I hacked into Apple, Microsoft, and dozens of other companies (Medium)](https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610)  
```

Comment:
All other are supply chain attacks incidents but this one is a kind of supply chain attack, it's not coherent

Reply:
It was based on this article https://www.reversinglabs.com/blog/ci/cd-security-breaches-update-software-security-approach to build which subject we are going to talk about.

It reference this case https://www.reversinglabs.com/blog/npm-dependency-confusion-hacks-target-german-firms#:~:text=%2D%20May%2011%2C%202022%3A,and%20industrial%20firms%20in%20Germany.
So maybe i should talk about this specific case on german firms, what do you think?